backend/apps/schemes/tests/test_service_transactions.py

```
@@ -0,0 +1,309 @@
+
+    def test_scheme_update_rollback_on_duplicate_error(self):
+        """
+        Test that scheme_update rolls back when duplicate is detected.
+
+        Scenario:
+        1. Create two schemes
+        2. Attempt to update scheme2 with scheme1's card code
+        3. Verify scheme2 data unchanged (rollback succeeded)
+        """
+        # Create first scheme
+        scheme1 = SchemeService.scheme_create(
```

Ruff (0.14.2)

Local variable `scheme1` is assigned to but never used

Remove assignment to unused variable `scheme1`

(F841)